### ISSUE TITLE
Yarn lock file not updated after lerna publish

### ISSUE THREAD

**Issue by rbirkgit** [lerna/lerna#3551]
## Current Behavior

I have a repo with multiple projects (workspaces). After `lerna publish` everything is updated and published as it should, except for the yarn lock file. 

We have projects having dependencies on other projects in the same repo. lerna updates them in the package.json file. With yarn 3.x these dependencies are reflected in the lock file. 

This means that when we later try building our repo, after a lerna publish, it fails. This because our build does `yarn install --frozen-lockfile` and then gets an error that the lock file would be modified. 

## Expected Behavior

If lerna updates the package,json files (like it does for me with the package dependencies), it also needs to make sure the lock file gets updated. Now I have to manually run `yarn install` and manually push a new changeset to the repo after every time we run lerna publish.

## Steps to Reproduce

Have a project depend on another project with yarn 3.x

e.g.
```
  "dependencies": {
    "@me/proj1": "~1.1.0",
```

Update `@me/proj1` then `lerna publish`. The version dependency gets updated by lerna in package.json but not the yarn.lock file.

### Failure Logs / Configuration

After yarn publish we get this when building our repo:

```
The lockfile would have been modified by this install, which is explicitly forbidden.
```

### Environment

lerna notice cli v6.5.1
lerna info versioning independent

 Environment info:

```
  System:
    OS: Windows 10 10.0.22000
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 18.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 3.2.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 9.3.1 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.37.3.
```

**Comment by ghiscoding** [comment]
just a side note on the command, I'm not using Yarn but I think it might be better with `yarn install --mode update-lockfile` 
as per their [mode](https://yarnpkg.com/cli/install#options-mode%20%230) description: 
> `update-lockfile` will skip the link step altogether, and only fetch packages that are missing from the lockfile (or that have no associated checksums). This mode is typically used by tools like Renovate or Dependabot to keep a lockfile up-to-date without incurring the full install cost.

As an alternative, you might be able to run the command via the `publish` (or `postpublish`) lifecycles that can be added to the root, it's not ideal because it won't be part of the git changes but it's one way of doing it. Ideally it would be better handled by the lib

**Comment by rbirkgit** [comment]
The point with ci pipelines is that the build should use exactly what is in the lock file. If someone by mistake checked in an updated package.json file without updated lock file, that means they never tested the code with the changes. We don't want that in the build and why the frozen-lockfile parameter. 

Anyway whatever the opinions are about this whether you agree or not :) it's a separate subject.

**Comment by fahslaj** [comment]
Hi @rbirkgit , thank you for reporting this issue. It has been fixed in PR #3555 and will be available when the next patch version of Lerna is released. I'll follow-up when this is done.

**Comment by fahslaj** [comment]
Hi everyone, this fix has been released in version 6.6.0!
